supertokens_python/recipe/session/recipe_implementation.py

```diff
@@ -221,74 +200,62 @@ async def get_session(
             validate_access_token_structure(
                 access_token_obj.payload, access_token_obj.version
             )
-        except Exception as e:
-            log_debug_message(
-                "getSession: Returning UNAUTHORISED because parsing failed"
-            )
-            return GetSessionUnauthorizedErrorResult(e)
-
-        try:
-            response = await session_functions.get_session(
-                self,
-                access_token_obj,
-                anti_csrf_token,
-                (anti_csrf_check is not False),
-                (check_database is True),
-            )
-        except Exception as e:
-            if isinstance(e, UnauthorisedError):
+        except Exception:
+            if session_required is False:
                 log_debug_message(
-                    "getSession: Returning TRY_REFRESH_TOKEN_ERROR because of an exception during get_session"
+                    "getSession: Returning undefined because parsing failed and session_required is False"
                 )
-                return GetSessionUnauthorizedErrorResult(e)
+                return None
 
-            log_debug_message(
-                "getSession: Returning UNAUTHORISED because of an exception during get_session"
-            )
-            return GetSessionUnauthorizedErrorResult(e)
+            raise UnauthorisedError("Token parsing failed", clear_tokens=False)
+
+        response = await session_functions.get_session(
+            self,
+            access_token_obj,
+            anti_csrf_token,
+            (anti_csrf_check is not False),
+            (check_database is True),
+        )
 
         log_debug_message("getSession: Success!")
 
-        if "accessToken" in response:
+        if response.accessToken is not None:
             payload = parse_jwt_without_signature_verification(
-                response["accessToken"]["token"]
+                response.accessToken.token
             ).payload
-            access_token_str = response["accessToken"]["token"]
-            expiry_time = response["accessToken"]["expiry"]
+            access_token_str = response.accessToken.token
+            expiry_time = response.accessToken.expiry
             access_token_updated = True
         else:
             payload = access_token_obj.payload
             access_token_str = access_token
-            expiry_time = response["session"]["expiryTime"]
+            expiry_time = response.session.expiryTime
             access_token_updated = False
 
         session = Session(
             self,
             self.config,
             access_token_str,
-            build_front_token(response["session"]["userId"], expiry_time, payload),
+            # FIXME: expiry time cannot be None
```

Comment:
Fix after @porcellus concludes what should be done (and maybe updates node SDK)

Reply:
The fixme comment is still left I think